internal/prober/multihttp/script.go

```diff
@@ -0,0 +1,264 @@
+
+	return template.JSEscapeString(u.String())
+}
+
+func buildHeaders(headers []*sm.HttpHeader) string {
+	var buf strings.Builder
+	buf.WriteRune('{')
+	for i, header := range headers {
+		if i > 0 {
+			buf.WriteRune(',')
+		}
+		buf.WriteRune('\'')
+		buf.WriteString(template.JSEscapeString(header.Name))
+		buf.WriteString(`':'`)
+		buf.WriteString(template.JSEscapeString(header.Value))
+		buf.WriteRune('\'')
+	}
+	buf.WriteRune('}')
+	return buf.String()
+}
+
+type assertionCondition sm.MultiHttpEntryAssertionConditionVariant
+
+func (c assertionCondition) Name(w *strings.Builder, subject, value string) {
+	w.WriteString(template.JSEscapeString(subject))
+	w.WriteRune(' ')
+
+	switch sm.MultiHttpEntryAssertionConditionVariant(c) {
+	case sm.MultiHttpEntryAssertionConditionVariant_NOT_CONTAINS:
+		w.WriteString(`does not contain`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_CONTAINS:
+		w.WriteString(`contains`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_EQUALS:
+		w.WriteString(`equals`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_STARTS_WITH:
+		w.WriteString(`starts with`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_ENDS_WITH:
+		w.WriteString(`ends with`)
+
+	}
+
+	w.WriteRune(' ')
+	w.WriteString(template.JSEscapeString(value))
+}
+
+func (c assertionCondition) Render(w *strings.Builder, subject, value string) {
+	switch sm.MultiHttpEntryAssertionConditionVariant(c) {
+	case sm.MultiHttpEntryAssertionConditionVariant_NOT_CONTAINS:
+		w.WriteRune('!')
+		fallthrough
+
+	case sm.MultiHttpEntryAssertionConditionVariant_CONTAINS:
+		w.WriteString(subject)
+		w.WriteString(`.includes('`)
+		w.WriteString(template.JSEscapeString(value))
+		w.WriteString(`')`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_EQUALS:
+		w.WriteString(subject)
+		w.WriteString(` === '`)
+		w.WriteString(template.JSEscapeString(value))
+		w.WriteString(`'`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_STARTS_WITH:
+		w.WriteString(subject)
+		w.WriteString(`.startsWith('`)
+		w.WriteString(template.JSEscapeString(value))
+		w.WriteString(`')`)
+
+	case sm.MultiHttpEntryAssertionConditionVariant_ENDS_WITH:
+		w.WriteString(subject)
+		w.WriteString(`.endsWith('`)
+		w.WriteString(template.JSEscapeString(value))
+		w.WriteString(`')`)
+	}
+}
+
+// buildChecks takes a single assertion and produces the corresponding JavaScript code.
+//
+// This function is a mess because a single assertion represents multiple types
+// of checks.
+func buildChecks(assertion *sm.MultiHttpEntryAssertion) string {
+	var b strings.Builder
+
+	b.WriteString(`check(response, { '`)
+
+	switch assertion.Type {
+	case sm.MultiHttpEntryAssertionType_TEXT:
+		cond := assertionCondition(assertion.Condition)
+
+		switch assertion.Subject {
+		case sm.MultiHttpEntryAssertionSubjectVariant_RESPONSE_BODY:
+			cond.Name(&b, "body", assertion.Value)
+			b.WriteString(`': response => `)
+			cond.Render(&b, "response.body", assertion.Value)
```

Comment:
this needs to perform variable replacements.

if the input looks like `${foo}` it needs to be replaced by `vars['foo']` so that variables are used...